diff --git a/node_modules/@types/uikit/index.d.ts b/node_modules/@types/uikit/index.d.ts
index d2c5c23..976b76a 100755
--- a/node_modules/@types/uikit/index.d.ts
+++ b/node_modules/@types/uikit/index.d.ts
@@ -12,16 +12,16 @@ type UIkitNodes = NodeList | HTMLCollection | UIkitNode;
 type UIkitNode = Node;
 
 export namespace UIkit {
-    const util: object;
-    const component: object;
+    const util: unknown;
+    const component: (name: string, options?: unknown) => unknown;
     const data: string;
     const prefix: string;
     const options: object;
     const version: string;
-    const use: object;
-    const mixin: object;
-    const extend: object;
-    const update: object;
+    const use: (plugin: unknown) => unknown;
+    const mixin: (mixin: object, component: string | ReturnType<typeof UIkit.component>) => void;
+    const extend: (options: object) => unknown;
+    const update: (element: UIkitElement, e: unknown) => void;
 
     interface UIkitAccordionOptions {
         active?: number | undefined;
@@ -318,6 +318,7 @@ export namespace UIkit {
     }
 
     interface UIkitTabElement {
+        index(): number;
         show(index: string | number | UIkitNode): void;
     }
 
