diff --git a/node_modules/@types/uikit/dist/js/uikit.d.ts b/node_modules/@types/uikit/dist/js/uikit.d.ts
index 6bab926..0130226 100755
--- a/node_modules/@types/uikit/dist/js/uikit.d.ts
+++ b/node_modules/@types/uikit/dist/js/uikit.d.ts
@@ -8,15 +8,15 @@ export interface Plugin {
 }
 
 export namespace UIkit {
-    const util: object;
-    const component: object;
+    const util: unknown;
+    const component: (name: string, options?: unknown) => unknown;
     const data: string;
     const prefix: string;
     const options: object;
     const version: string;
-    const mixin: object;
-    const extend: object;
-    const update: object;
+    const mixin: (mixin: object, component: string | ReturnType<typeof UIkit.component>) => void;
+    const extend: (options: object) => unknown;
+    const update: (element: UIkitElement, e: unknown) => void;
 
     function use(plugin: Plugin): typeof UIkit;
 
@@ -334,6 +334,7 @@ export namespace UIkit {
     }
 
     interface UIkitTabElement {
+        index(): number;
         show(index: string | number | UIkitNode): void;
     }
 
@@ -351,6 +352,7 @@ export namespace UIkit {
 
     interface UIkitToggleElement {
         toggle(): void;
+        isToggled(): boolean;
     }
 
     type Toggle = (element: UIkitElement, options?: UIkitToggleOptions) => UIkitToggleElement;
